qlib/backtest/report.py

```diff
@@ -560,3 +553,236 @@ def get_trade_indicator(self):
+            elif isinstance(other, PandasOrderIndicator.SingleMetric):
+                return PandasOrderIndicator.SingleMetric(other.metric + self.metric)
+            else:
+                return NotImplemented
+
+        def __sub__(self, other):
+            if isinstance(other, (int, float)):
+                return PandasOrderIndicator.SingleMetric(self.metric - other)
+            elif isinstance(other, PandasOrderIndicator.SingleMetric):
+                return PandasOrderIndicator.SingleMetric(self.metric - other.metric)
+            else:
+                return NotImplemented
+
+        def __rsub__(self, other):
+            if isinstance(other, (int, float)):
+                return PandasOrderIndicator.SingleMetric(other - self.metric)
+            elif isinstance(other, PandasOrderIndicator.SingleMetric):
+                return PandasOrderIndicator.SingleMetric(other.metric - self.metric)
+            else:
+                return NotImplemented
+
+        def __mul__(self, other):
+            if isinstance(other, (int, float)):
+                return PandasOrderIndicator.SingleMetric(self.metric * other)
+            elif isinstance(other, PandasOrderIndicator.SingleMetric):
+                return PandasOrderIndicator.SingleMetric(self.metric * other.metric)
+            else:
+                return NotImplemented
+
+        def __truediv__(self, other):
+            if isinstance(other, (int, float)):
+                return PandasOrderIndicator.SingleMetric(self.metric / other)
+            elif isinstance(other, PandasOrderIndicator.SingleMetric):
+                return PandasOrderIndicator.SingleMetric(self.metric / other.metric)
+            else:
+                return NotImplemented
+
+        def __eq__(self, other):
+            if isinstance(other, (int, float)):
+                return PandasOrderIndicator.SingleMetric(self.metric == other)
+            elif isinstance(other, PandasOrderIndicator.SingleMetric):
+                return PandasOrderIndicator.SingleMetric(self.metric == other.metric)
+            else:
+                return NotImplemented
+
+        def __gt__(self, other):
+            if isinstance(other, (int, float)):
+                return PandasOrderIndicator.SingleMetric(self.metric < other)
+            elif isinstance(other, PandasOrderIndicator.SingleMetric):
+                return PandasOrderIndicator.SingleMetric(self.metric < other.metric)
+            else:
+                return NotImplemented
+
+        def __lt__(self, other):
+            if isinstance(other, (int, float)):
+                return PandasOrderIndicator.SingleMetric(self.metric > other)
+            elif isinstance(other, PandasOrderIndicator.SingleMetric):
+                return PandasOrderIndicator.SingleMetric(self.metric > other.metric)
+            else:
+                return NotImplemented
+        
+        def __len__(self):
+            return len(self.metric)
+
+        def sum(self):
```

Comment:
these functions are relied on by the framework.
But there is no corresponding interface.